packages/astro/src/default/utils/content.ts

```diff
@@ -75,250 +208,226 @@ export async function getTutorial(): Promise<Tutorial> {
         id: lessonId,
         filepath: id,
         order: -1,
-        part: {
-          id: partId,
-          title: _tutorial.parts[partId].data.title,
-        },
-        chapter: {
-          id: chapterId,
-          title: _tutorial.parts[partId].chapters[chapterId].data.title,
-        },
         Markdown: Content,
         slug: getSlug(entry),
         files,
         solution,
       };
 
-      lessons.push(lesson);
+      if (partId) {
+        if (!tutorial.parts[partId]) {
+          throw new Error(`Could not find part '${partId}'`);
+        }
+
+        lesson.part = {
+          id: partId,
+          title: tutorial.parts[partId].data.title,
+        };
+      }
 
-      _tutorial.parts[partId].chapters[chapterId].lessons[lessonId] = lesson;
+      if (partId && chapterId) {
+        if (!tutorial.parts[partId].chapters[chapterId]) {
+          throw new Error(`Could not find chapter '${chapterId}'`);
+        }
+
+        lesson.chapter = {
+          id: chapterId,
+          title: tutorial.parts[partId].chapters[chapterId].data.title,
+        };
+      }
```

Comment:
I'm wondering if we should do this in a second pass in case parts, chapters and lessons are not in order.

It's also something that could have been an issue with the previous logic though so maybe we don't need to care about this.

Reply:
At this point they should be ordered:

https://github.com/stackblitz/tutorialkit/blob/0b2896a0e9e63a74ce8a3775523e9592e4b6b70f/packages/astro/src/default/utils/content.ts#L20-L22

https://github.com/stackblitz/tutorialkit/blob/0b2896a0e9e63a74ce8a3775523e9592e4b6b70f/packages/astro/src/default/utils/content.ts#L271-L278

But yes, it's not visible from this function only. Should we move the `sortCollection` function call inside this function, so that we can be sure it's always sorted before looping it?